worker-courses/domain/get-all-courses.ts

```diff
@@ -0,0 +1,17 @@
+import { UdacityFetcher } from '../providers/udacity-fetcher';
+import { CourseRepository } from '../db/mongo-db.connector';
+import { Course } from '../models/course';
+
+export class GetAllCoursesUseCase {
+
+  constructor() {}
+
+  async getAllCourses(): Promise<any>  {
+    let udacityFetcher = new UdacityFetcher();
+
+    udacityFetcher.getAllCourses().then(function (courses: Course[]){
+      return CourseRepository.update(courses);
+  });
```

Comment:
misaligned, additionally, you could simplify this way:

```typescript
udacityFetcher.getAllCourses().then(courses => CourseRepository.update(courses));
```

Additionally, it makes no sense to use `async` keyword if you're not returning any value. I believe what you meant was:

```typescript
return udacityFetcher.getAllCourses().then(courses => CourseRepository.update(courses));
```